### PR TITLE
modified show courses and displaying info

### PR DESCRIPTION
**Pull request by eleensmathew**
## Description

adding course. Modified models and modified views and html pages for displaying coures.

## Related Issue

Fixes # ([issue](https://github.com/ContriHUB/Youdemy/issues/4))

### CODE REVIEW

**Comment by eleensmathew** [comment]
Hi @kunal2812  Im facing some difficulty in making an @add_course view because i understand that multiple titles and multiple chapters can be added on the same page. I'm not too sure how to do this. Can you provide some resources how multiple chapters and each chapters titles can be saved in the view?

**Comment by kunal2812** [comment]
You are facing issues in creating schema for the course or saving the entries to the tables ?

**Comment by eleensmathew** [comment]
the html page is supposed to save mutliple titles and multiple chapters so im having trouble extracting that information from the html in the view. Can i modify the page so that only one title/one chapter can be added at a time?

**Comment by kunal2812** [comment]
Can you share your schema design ?

**Comment by eleensmathew** [comment]
You can see it here https://github.com/ContriHUB/Youdemy/blob/2f1a91de0272e102018cd21ecb7b68264e8f5439/main/models.py

**Comment by kunal2812** [comment]
See, a course can have multiple chapters, so you can create a table called 'Chapters' with columns such as | chapter_id | course_id | content |. Similarly, you can create a table for titles. Whenever you add a new chapter, you'll click on the course link or something to retrieve the course_id

**Comment by kunal2812** [comment]
I'll share some resources for that.

**Comment by eleensmathew** [comment]
Yes i have made the tables like that only. I am facing a problem for the add course page where the mentor can choose to add 1 chapter with one title or even 5 chapter with 5 titles each. I am unable to figure out how to send that data to the view and have that data cleaned.

**Comment by kunal2812** [comment]
Check this for now. Will link more.

https://www.youtube.com/watch?v=c2hbT0uIcOQ
https://www.youtube.com/watch?v=PICYTJqj__o
https://www.youtube.com/watch?v=s3T-w2jhDHE

**Comment by kunal2812** [comment]
Check these out

**Comment by eleensmathew** [comment]
Hey can I have an extra day? i Have an interview tomorrow so gotta prepare

**Comment by kunal2812** [comment]
Sure. All the best.

**Comment by eleensmathew** [comment]
Hi i added all the features but now im debugging the code.

**Comment by eleensmathew** [comment]
Hey @kunal2812 th ecode is working. Should I add the option to delete and edit course as well? right now only adding courses ,chapters , titles and publishing courses can be done by mentors.

**Comment by kunal2812** [comment]
Yes you can work on it.

**Comment by eleensmathew** [comment]
Hey its done. If you have any comments let me know.

**Comment by kunal2812** [comment]
Great frontend skills, work you've done looks neat.

Has some minor issues will list them for you:

image

- Info about user not passed to the /courses url so getting user as "None"

- Once the course is published are you not allowing the course mentor to edit it ? It should be editable then too right ? What do you suggest ?

image

- Fix the navbar on homepage

image

- Make these more descriptive. For eg. "Course Thumbnail", "Chapter Thumbnail", "Title Thumbnail"

- Is course section removed from the homepage ? I added a course but still can't see it there.

image

- Also even though I'm selecting images to upload as thumbnail still I can't seem them (Same as well for chapters)

image

- You probably missed out this "Questions" part.

**Comment by eleensmathew** [comment]
Ok i will work on them as soon as I can
Eleena

**Comment by eleensmathew** [comment]
I think i resolved all of them but unfortunately I'm still unsure why the pictures aren't getting saved? I changed the location from static folder to media but im getting errors maybe because of my migrations? Can you check it out and let me know? @kunal2812

**Comment by kunal2812** [comment]
https://www.youtube.com/watch?v=O5YkEFLXcRg

You can refer to this tutorial to learn to upload images in Django. Also you can look into the Blog feature in the Sheher repository to gain more insight on that.

Also most of the issues mentioned above aren't fixed yet. Nav Bar on home page is still not fixed and image fields aren't made descriptive yet.

**Comment by eleensmathew** [comment]
Sure thanks, will fix issues soon
Eleena

**Comment by eleensmathew** [comment]
what exactly should i fix in the nav bar?

**Comment by kunal2812** [comment]
Frontend is a bit off

How it is now (In Home page and Courses page):
image

How it should look like:
image

**Comment by eleensmathew** [comment]
I actually added 'Edit your courses' in the nav bar for mentors so they can
access it. Do you want to decrease the padding on the sides or remove the
feature from the navbar all together?

**Comment by eleensmathew** [comment]
If the issue is uniformity of navbar i have fixed it.

**Comment by kunal2812** [comment]
Yes, the problem was with the uniformity of the nav bar.

**Comment by eleensmathew** [comment]
Screenshot 2023-10-23 022746
Is this okay? Or should i alter the choose file part as well?

**Comment by kunal2812** [comment]
Yes this will work

**Comment by eleensmathew** [comment]
I think I fixed all the issues now. Can't believe i'm working on the same issue for a week now. Anyway let me know what you think. If you think I can add something more to get additional points I would love to hear it.

**Comment by kunal2812** [comment]
Course functionality is looking good. 
I see you have added answers to the questions for titles, but are they being stored in a database? And if they are stored, can you make them visible when I click on the question?

image

Nav bar is still not fixed.

image

image

This is the problem.

Also, remember that the nav-link was active based on the page we were on. Now, since you have generalized the navbar, this has been removed. It is a minor issue but can be fixed. You can either revert the navbar changes or apply the 'nav-item active' class based on the URL

```
<li class="nav-item active">
       <a href="{% url 'home' %}" class="nav-link">Home</a>
</li>
// Now home remains active on all pages
```

Also, do not remove the courses section from the home page. Courses should still be present on the home page, as they were earlier.

This issue is taking longer than expected. Help me in closing it as soon as possible. Will be adding more issues as soon as this one is resolved so keep an eye out for them.

**Comment by eleensmathew** [comment]
image
this is the updated navbar which i hope is fine? I'm not sure about what you mean by courses in the navbar though.

**Comment by eleensmathew** [comment]
I don't think i removed the courses section from the index. i've checked the files changed section so i don't think i've removed a courses section? please let me know what you mean exactly. I have made all the other changes.

**Comment by kunal2812** [comment]
> image this is the updated navbar which i hope is fine? I'm not sure about what you mean by courses in the navbar though.

image

Its still looking like this at my end.

**Comment by kunal2812** [comment]
> I don't think i removed the courses section from the index. i've checked the files changed section so i don't think i've removed a courses section? please let me know what you mean exactly. I have made all the other changes.

Checkout to master branch.

image

I can see Courses section here.

File: dbsql
Comment (kunal2812) [inline dbsql:1]: Remove this from commit.

**Comment by eleensmathew** [comment]
image
I added it.

File: dbsql
Comment (eleensmathew) [reply dbsql:1]: done

**Comment by kunal2812** [comment]
Is navbar fixed ?

**Comment by eleensmathew** [comment]
image

its looking like this to me so im not sure why its appearing different for you

**Comment by eleensmathew** [comment]
i have made changes in styles.css to fix the navbar problem

**Comment by kunal2812** [comment]
Commit all the changes. I'll check again

**Comment by eleensmathew** [comment]
I have commited all the changes already.

**Comment by kunal2812** [comment]
It looks okay when I view it in the Edge browser, but in Chrome, it is not how it should be. Make changes for Chrome as well.

Edge
image

Chrome
image

**Comment by eleensmathew** [comment]
I am getting the same display in chrome as well. Can you try clearing the
cache? anyway let me see if i can fix it.

**Comment by kunal2812** [comment]
Yeah. Clearing the cache worked. Reviewing now.

**Comment by kunal2812** [comment]
Courses on "Course" page:
image

Courses on "Home" page:
image

Course language should be visible on "Home" page as well.

**Comment by eleensmathew** [comment]
> Courses on "Course" page: image
> 
> Courses on "Home" page: image
> 
> Course language should be visible on "Home" page as well.

fixed

**Comment by kunal2812** [comment]
image

Fix breadcrumbs in the course page

image

When I'm on any course "Course" should be highlighted

**Comment by eleensmathew** [comment]
Done.

**Comment by kunal2812** [comment]
image

No option to add questions when the course has been published.

**Comment by kunal2812** [comment]
Also not able to add chapters. Neither when the course is unpublished nor when it is.

**Comment by eleensmathew** [comment]
the current course looks like this 
image
and 
image
i have fixed all the highlighted errors. 🫠

**Comment by kunal2812** [comment]
> image
> 
> When I'm on any course "Course" should be highlighted

This one ?

**Comment by eleensmathew** [comment]
Yes fixed that as well 
image

**Comment by eleensmathew** [comment]
ps i deserve 6 billion points

**Comment by kunal2812** [comment]
image

I think title thumbnail field is redundant here. Or is it used somewhere ?

**Comment by eleensmathew** [comment]
It is redundant. Do you want me to remove it?
Eleena

**Comment by kunal2812** [comment]
Yes

**Comment by eleensmathew** [comment]
Done

**Comment by kunal2812** [comment]
image

User with "Student" role is getting these options as well 😶

**Comment by eleensmathew** [comment]
Changed that 
image
also added title edit 
image

**Review by kunal2812 — APPROVED** [review]
Looking great !!

**Pull request merged**